src/Symfony/Component/Workflow/Workflow.php

```diff
@@ -185,32 +190,41 @@ public function getMarkingStore()
         return $this->markingStore;
     }
 
-    private function doCan($subject, Marking $marking, Transition $transition)
+    private function isTransitionEnabled($subject, Marking $marking, Transition $transition, $throwException = false): bool
```

Comment:
exceptions are not necessary imho. always returing a (possibly empty) TransitionBlockerList ist much more straight forward. in the calling place you can do `$this->isTransitionEnabled(...)->isEnabled()`. `isEnabled` is then a new helper method on TransitionBlockerList. maybe renaming the method to `determineTransitionAvailablity` or `computeTransitionState` clarifies the intention: `$this->computeTransitionState(...)->isEnabled()`. Another helper method `isAvailable` could be added to TransitionBlockerList, that returns true, either if `isEnabled()` returns true or if all blockers on the list are of type `BLOCKED_BY_MARKING`

Reply:
You are right. I will remove the `throwException` parameter.
But I would like to avoid the word "state" as it could be confusing (place / state / marking)
About isAvailable is don't think is really useful. We will see